src/sources/kafka.rs

```diff
@@ -439,15 +509,66 @@ impl ReceivedMessage {
         }
     }
 
-    fn apply(&self, keys: &Keys<'_>, event: &mut Event) {
+    fn apply(&self, keys: &Keys<'_>, event: &mut Event, log_namespace: LogNamespace) {
         if let Event::Log(ref mut log) = event {
-            log.insert(keys.source_type, Bytes::from("kafka"));
-            log.insert(keys.timestamp, self.timestamp);
-            log.insert(keys.key_field, self.key.clone());
-            log.insert(keys.topic, Value::from(self.topic.clone()));
-            log.insert(keys.partition, Value::from(self.partition));
-            log.insert(keys.offset, Value::from(self.offset));
-            log.insert(keys.headers, Value::from(self.headers.clone()));
+            if let LogNamespace::Vector = log_namespace {
+                // We only want an auto-generated timestamp in the Vector namespace.
+                log_namespace.insert_standard_vector_source_metadata(
+                    log,
+                    KafkaSourceConfig::NAME,
+                    Utc::now(),
+                );
+            } else {
+                log.insert("source_type", KafkaSourceConfig::NAME);
```

Comment:
It looks like this was using `schema.source_type_key()` previously. Why did it change?

Reply:
The field isn't configurable by the user, so I dropped the source_type field from `keys`. I'm happy to change this to `log_schema().source_type_key()` if that's more correct.

Reply:
`schema.source_type_key()` was being pulled from the `log_schema()` before, just in a more roundabout way than normal. It should still be configurable in the Legacy namespace 👍

Reply:
Good callout. I actually didn't know that `log_schema()` was configurable.